streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -63,10 +66,21 @@ public RecordCollector(Producer<byte[], byte[]> producer) {
         this.offsets = new HashMap<>();
     }
 
-    public <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer) {
+    public void send(ProducerRecord<byte[], byte[]> record) {
```

Comment:
I feel this optimization is not necessarily since calling ```ByteArraySerializer.serialize()``` does not introduce much overhead, but requires other instantiations of this class to override two functions.